conans/test/utils/genconanfile.py

```diff
@@ -281,6 +291,16 @@ def _package_method(self):
                           '            os.symlink(os.path.basename("{}"), '
                           'os.path.join(self.package_folder, "{}"))'.format(key, key, value)
                           for key, value in self._package_files_link.items()])
+        if self._package_folders:
+            lines.extend(['        tools.mkdir(os.path.join(self.package_folder, "{}"))'
+                          ''.format(folder) for folder in self._package_folders])
+        if self._package_folders_link:
```

Comment:
@danimtb 

I would say that this is an overly test-specific thing, that this should be implemented directly in the test, leaving ``GenConanfile()`` more generic. Only common conanfile patterns should be here.

It is true that the ``_package_files_link`` was already there, but that pattern is used just exactly once in the whole codebase, so probably that deserved to be specified in the test directly, and not generalized here.

Asking @jgsogo for a second opinion.

Reply:
Yes, I think the same. `GenConanfile` must be something generic... even if it cannot be used for some tests. Sometimes it is also better to write the plain `conanfile` source if it helps to understand what we are testing.

Reply:
...but this argument also applies to the `with_package_file(..., link)` that is already in the `GenConanfile`, only used once.

Reply:
Yeah, this is what I meant with:

> It is true that the _package_files_link was already there, but that pattern is used just exactly once in the whole codebase, so probably that deserved to be specified in the test directly, and not generalized here.

``package_files_link`` shouldn't be in GenConanfile, but in the specific test that is using it.

Reply:
done!